src/middleware/express-error.js

```diff
@@ -0,0 +1,95 @@
+const uuid = require('uuid');
+const debug = require('debug')('api:errors');
+const ErrorReply = require('../error-reply');
+
+/**
+ * Create parameter validation middle-ware instance, given a mapping from
+ * parameter to regular expression or function that returns a message as string
+ * if the parameter is invalid.
+ *
+ * Parameters not listed in `req.params` will be ignored. But parameters
+ * present must match the pattern given in `options` or the request will be
+ * rejected with a 400 error message.
+ */
+const expressError = ({errorCodes, entry, monitor}) => {
+  const {name: method, cleanPayload} = entry;
+  return (err, req, res, next) => {
+
+    const incidentId = uuid.v4();
+    if (!(err instanceof ErrorReply)) {
+      err = new ErrorReply({
+        code: 'InternalServerError',
+        message: 'Internal Server Error, incidentId: ' + incidentId,
+        details: {incidentId}
+      });
+      debug(
+        'Error occurred handling: %s, err: %s, as JSON: %j, incidentId: %s',
+        req.url, err, err, incidentId, err.stack
+      );
+      if (monitor) {
+        err.incidentId = incidentId;
+        err.method = method;
+        err.params = req.params;
+        let payload = req.body;
+        if (cleanPayload) {
+          payload = cleanPayload(payload);
+        }
+        err.payload = req.payload;
+        monitor.reportError(err, {method});
+      }
+    }
+
+    let code = err.code || 'InternalServerError';
+    let details = err.details || {incidentId};
```

Comment:
There's no need for these `|| ..` anymore -- `err` is always an ErrorReply, and ErrorReply will always have these properties.